lib/connectors/memory.js

```diff
@@ -207,18 +207,27 @@ Memory.prototype.create = function create(model, data, options, callback) {
   var idName = this.idName(model);
   id = (props[idName] && props[idName].type && props[idName].type(id)) || id;
   this.setIdValue(model, data, id);
-  if(!this.collection(model)) {
+  if (!this.collection(model)) {
     this.collection(model, {});
   }
 
-  if (this.collection(model)[id]) {
-    return process.nextTick(function() {
-      callback(new Error('Duplicate entry for ' + model + '.' + idName));
-    });
-  }
+  if (this.collection(model)[id])
+    return callback(new Error('Duplicate entry for ' + model + '.' + idName));
```

Comment:
Why did you remove the process.nextTick wrapper which ensures the callback to invoked after this function returns?